### PR TITLE
feat(dock-manager): add styling topic

### LINKED ISSUE #158: Add a CSS parts section to DockManager topic
Document the new feature for elements' customizations [Issue #3](https://github.com/IgniteUI/igniteui-dockmanager/issues/3)

### PR DESCRIPTION
**Pull request by zhosti**
Closes #158

### CODE REVIEW

File: doc/en/components/dock-manager.md [hunk 1]
```diff
@@ -373,6 +373,60 @@ The shortcuts are as follows:
 
 Practice all of the above mentioned actions in the sample [`demo`](dock-manager.md#$Platform$-dock-manager-example).
 
+## Styling
+
+The Dock Manager use a sahadow DOM to encapsulate his styles and behaviors. As a result, you can't simply target their internals with the usual CSS selectors. That is why we expose components `parts` that can be targeted with the `::part` CSS selector.
```
Comment (gedinakova) [inline doc/en/components/dock-manager.md:378]: The Dock Manager use**s** a shadow ....

Comment (gedinakova) [reply doc/en/components/dock-manager.md:378]: As a result, you can't simply target its internal elements with ....

File: doc/en/components/dock-manager.md [hunk 2]
```diff
@@ -373,6 +373,60 @@ The shortcuts are as follows:
...
+
+```css
+igc-dockmanager::part(content-pane) {
+  border-radius: 10px;
+}
+```
+
+In the following example, we will demonstrate the ability of customizing the Dock Manager through some of the exposed parts that we've provided.
```
Comment (gedinakova) [inline doc/en/components/dock-manager.md:386]: In the following example, we demonstrate the ability of customizing the Dock Manager through some of the CSS parts that we've exposed.

File: doc/en/components/dock-manager.md [hunk 3]
```diff
@@ -373,6 +373,60 @@ The shortcuts are as follows:
...
+`root-docking-indicator` | The root docking indicator.
+`pane-navigator` | The pane navigator component.
+`pane-navigator-header` | The header area of the pane navigator.
+`pane-navigator-body` | The body area of the pane navigator.
+`pane-navigator-items-group` | An items group in the pane navigator component.
+`pane-navigator-items-group-title` | The title element of an items group in the pane navigator.
+`pane-navigator-item` | An item in the pane navigator.
+
+
 ## Themes
```
Comment (gedinakova) [inline doc/en/components/dock-manager.md:430]: Maybe we could make Themes a sub category of styling

**Review by gedinakova — CHANGES_REQUESTED** [review]

File: doc/en/components/dock-manager.md [hunk 4]
```diff
@@ -373,7 +373,61 @@ The shortcuts are as follows:
 
 Practice all of the above mentioned actions in the sample [`demo`](dock-manager.md#$Platform$-dock-manager-example).
 
-## Themes
+## Styling
+
+The Dock Manager uses a sahadow DOM to encapsulate his styles and behaviors. As a result, you can't simply target its internal elements with the usual CSS selectors. That is why we expose components `parts` that can be targeted with the `::part` CSS selector.
```
Comment (gedinakova) [inline doc/en/components/dock-manager.md:378]: 'sahadow' => 'shadow'

**Review by gedinakova — CHANGES_REQUESTED** [review]

**Review by gedinakova — APPROVED** [review]

File: doc/en/components/dock-manager.md [hunk 5]
```diff
@@ -373,7 +373,61 @@ The shortcuts are as follows:
...
+}
+```
+
+In the following example, we demonstrate the ability of customizing the Dock Manager through some of the CSS parts that we've exposed.
+
+<code-view style="height: 600px"
+           data-demos-base-url="{environment:dvDemosBaseUrl}"
+           iframe-src="{environment:dvDemosBaseUrl}/layouts/dock-manager-styling"
+           alt="$Platform$ Dock Manager Styling Example"
+           github-src="layouts/dock-manager/styling">
```
Comment (HUSSAR-mtrela) [inline doc/en/components/dock-manager.md:392]: this sample is missing in Blazor repo, React repo, and Angular repo

https://github.com/IgniteUI/igniteui-blazor-examples/tree/vnext/samples/layouts/dock-manager

https://github.com/IgniteUI/igniteui-react-examples/tree/vnext/samples/layouts/dock-manager

https://github.com/IgniteUI/igniteui-angular-samples/tree/vNext/src/app/layouts/dock-manager/dock-manager-sample

**Review by HUSSAR-mtrela — CHANGES_REQUESTED** [review]

**Comment by HUSSAR-mtrela** [comment]
do not merge this PR - see comments in my review

File: doc/en/components/dock-manager.md [hunk 6]
```diff
@@ -325,15 +325,24 @@ this.dockManager.layout = { ...layout };
 
 The Dock Manager component raises events when specific end-user interactions are performed for example closing, pinning, resizing and dragging a pane. You can find the full list of Dock Manager events [here]({environment:infragisticsBaseUrl}/products/ignite-ui/dock-manager/docs/typescript/latest/interfaces/igcdockmanagereventmap.html). Here is how to add an event listener for the [`paneClose`]({environment:infragisticsBaseUrl}/products/ignite-ui/dock-manager/docs/typescript/latest/interfaces/igcdockmanagereventmap.html#paneclose) event:
 
+<!-- Blazor -->
+```razor
```
Comment (gedinakova) [inline doc/en/components/dock-manager.md:329]: I believe marking with ```razor is enough. However, are you sure about the code itself?

**Review by gedinakova — APPROVED** [review]

**Comment by gedinakova** [comment]
@HUSSAR-mtrela Martin, could you, please, check if this PR is ready to be merged now?

**Comment by gedinakova** [comment]
@HUSSAR-mtrela Hi Martin, we really need this PR merged. We've already released the package last week, so we have to demonstrate the CSS parts we've added and how they can use it to style the component. This was a feature requested by a few customers, so they are expecting it. Can you, please, check if everything is OK now?

**Review by HUSSAR-mtrela — APPROVED** [review]

**Pull request merged**